File: Testing/Code for Testing/Basic/JS/M005_objects_test.js
function createPoint() {
    return { x: 0, y: 0 };
}

class Animal {
    constructor(name = "", age = 0) {
        this.name = name;
        this.age = age;
    }
    speak() {
        return (this.name + " makes a sound");
    }
}

class Dog extends Animal {
    constructor(name = "", age = 0, breed = "") {
        super(name, age);
        this.breed = breed;
    }
    speak() {
        return (this.name + " barks");
    }
    fetch() {
        return (this.name + " fetches the ball");
    }
}

class Cat extends Animal {
    constructor(name = "", age = 0) {
        super(name, age);
    }
    speak() {
        return (this.name + " meows");
    }
}


function main() {
    let p = {x: 10, y: 20};
    console.log(p.x);
    console.log(p.y);
    let a = new Animal();
    a.name = "Generic";
    a.age = 5;
    console.log(a.speak());
    let d = new Dog();
    d.name = "Rex";
    d.age = 3;
    d.breed = "Labrador";
    console.log(d.speak());
    console.log(d.fetch());
    let c = new Cat();
    c.name = "Whiskers";
    c.age = 7;
    console.log(c.speak());
    return 0;
}

function __ul_sort(arr, n) { const sub = arr.slice(0, n).sort((a, b) => a < b ? -1 : a > b ? 1 : 0); for (let i = 0; i < n; i++) arr[i] = sub[i]; }

main();

File: Testing/Code for Testing/Basic/JS/M005_objects_test.py
from dataclasses import dataclass, field

def __ul_sort(arr, n): arr[:n] = sorted(arr[:n])

class __ul_Exception(Exception):
    def __init__(self, type_name, message):
        self.type_name = type_name
        self.message = message
        super().__init__(message)

@dataclass
class Point:
    x: int = 0
    y: int = 0

class Animal:
    def __init__(self, name="", age=0):
        self.name = name
        self.age = age
    def speak(self):
        return (self.name + ' makes a sound')

class Dog(Animal):
    def __init__(self, name="", age=0, breed=""):
        super().__init__(name, age)
        self.breed = breed
    def speak(self):
        return (self.name + ' barks')
    def fetch(self):
        return (self.name + ' fetches the ball')

class Cat(Animal):
    def __init__(self, name="", age=0):
        super().__init__(name, age)
    def speak(self):
        return (self.name + ' meows')


def main():
    p = Point(x=10, y=20)
    print(p.x)
    print(p.y)
    a = Animal()
    a.name = 'Generic'
    a.age = 5
    print(a.speak())
    d = Dog()
    d.name = 'Rex'
    d.age = 3
    d.breed = 'Labrador'
    print(d.speak())
    print(d.fetch())
    c = Cat()
    c.name = 'Whiskers'
    c.age = 7
    print(c.speak())
    return 0


if __name__ == "__main__":
    main()
